test/extensions/filters/http/common/fuzz/filter_fuzz_test.cc

```diff
@@ -0,0 +1,131 @@
+
+    Http::TestHeaderMapImpl headers = Fuzz::fromHeaders(data.headers());
+    if (data.data().size() == 0 && !data.has_trailers()) {
+      end_stream = true;
+    }
+    ENVOY_LOG_MISC(debug, "Decoding headers: {} ", data.headers().DebugString());
+    const auto& headersStatus = filter->decodeHeaders(headers, end_stream);
+    if (headersStatus != Http::FilterHeadersStatus::Continue ||
+        headersStatus != Http::FilterHeadersStatus::StopIteration) {
+      return;
+    }
+
+    for (int i = 0; i < data.data().size(); i++) {
+      if (i == data.data().size() - 1 && !data.has_trailers()) {
+        end_stream = true;
+      }
+      Buffer::OwnedImpl buffer(data.data().Get(i));
+      ENVOY_LOG_MISC(debug, "Decoding data: {} ", buffer.toString());
+      if (filter->decodeData(buffer, end_stream) != Http::FilterDataStatus::Continue) {
+      }
+    }
+
+    if (data.has_trailers()) {
+      ENVOY_LOG_MISC(debug, "Decoding trailers: {} ", data.trailers().DebugString());
+      Http::TestHeaderMapImpl trailers = Fuzz::fromHeaders(data.trailers());
+      filter->decodeTrailers(trailers);
+    }
+  }
+
+  // This creates the filter config and runs decode.
+  void fuzz(const envoy::extensions::filters::network::http_connection_manager::v3alpha::HttpFilter&
+                proto_config,
+            const test::fuzz::HttpData& data) {
+    try {
+      // Try to create the filter. Exit early if the config is invalid or violates PGV constraints.
+      auto& factory = Config::Utility::getAndCheckFactoryByName<
+          Server::Configuration::NamedHttpFilterConfigFactory>(proto_config.name());
+      ProtobufTypes::MessagePtr message = Config::Utility::translateToFactoryConfig(
+          proto_config, factory_context_.messageValidationVisitor(), factory);
+      Http::FilterFactoryCb cb =
+          factory.createFilterFactoryFromProto(*message, "stats", factory_context_);
+      cb(filter_callback_);
+    } catch (const EnvoyException& e) {
+      ENVOY_LOG_MISC(debug, "Controlled exception {}", e.what());
+      return;
+    }
+
+    decode(filter_.get(), data);
+  }
+
+  NiceMock<Server::Configuration::MockFactoryContext> factory_context_;
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks_;
+  NiceMock<Http::MockFilterChainFactoryCallbacks> filter_callback_;
+  std::shared_ptr<Http::StreamDecoderFilter> filter_;
+  NiceMock<Envoy::Network::MockConnection> connection_;
+  Network::Address::InstanceConstSharedPtr addr_;
+};
+
+DEFINE_PROTO_FUZZER(const test::extensions::filters::http::FilterFuzzTestCase& input) {
+  static PostProcessorRegistration reg = {
+      [](test::extensions::filters::http::FilterFuzzTestCase* input, unsigned int seed) {
+        // This ensures that the mutated configs all have valid filter names.
+        static const std::vector<absl::string_view> filter_names = Registry::FactoryRegistry<
+            Server::Configuration::NamedHttpFilterConfigFactory>::registeredNames();
```

Comment:
Looks like LPM is not very helpful with google.protobuf.Any
We should add support for that. https://github.com/google/libprotobuf-mutator/issues/151
For now you can either keep this post processor assuming it will not conflict with future LPM support or you can just remove it assuming fuzzing performance will improve in future on it's own.

Reply:
Actually we need somehow to find supported types for any without changing the proto definition. I guess this callback can be useful for that. I don't know for sure yet how this is going to work on LPM side.

Reply:
Yeah, I would guess it would essentially lookup the proto in the descriptor pool with type URL, unpack to it, and then do whatever is normally done by LPM, then repack. But, that's without any knowledge of LPM details. I think this is what we need to make this work efficiently, I know @asraa has been doing some experiments to see what can be done with the existing mutator, so there might be something useful even without that.

Reply:
So the message with embedded Any must have (1) valid name, (2) appropriate type_url based on name, (3) Any of the matching type. With the existing mutator what I've done is:
- After setting valid name, will LPM eventually create a non-trivial Any and appropriate type_url? -> Did not seem to after millions of runs
- After setting valid name, and adding valid type_urls and serialized protos to a dictionary, will LPM combine appropriately? Nope.
- After setting valid name and matching type_url, and adding serialized protos to a dictionary, will LPM populate the `value` field with the dictionary string? -> Yes! I got much more lucrative results.

So for now, I'm relying on dictionaries having full config strings pulled from unit test examples. I think we can iterate further later.